### ISSUE TITLE
How is the MAC address converted to an EUI?

### ISSUE THREAD

**Issue by 4refr0nt** [lorabasics/basicstation#45]
```
   ...
   "station_conf": {
        "routerid":  "0000e8eb1141xxxx",
        /*    "euiprefix": "",   */
   ...
```
```
...
2019-11-07 07:14:29.498 [SYS:INFO] proto EUI   : 0:e8eb:1141:xxxx       (station.conf)
2019-11-07 07:14:29.498 [SYS:INFO] prefix EUI  : ::1    (builtin)
2019-11-07 07:14:29.498 [SYS:INFO] Station EUI : e8eb:11ff:fe41:xxxx
...
2019-11-07 07:14:29.734 [TCE:INFO] Infos: e8eb:11ff:fe41:xxxx e8eb:11ff:fe41:xxxx ws://127.0.0.1:3001/gateway/e8eb11fffe41xxxx
...
```
This will caused to operate incorrect gataway MAC `e8eb11fffe41xxxx`, but real MAC is `0000e8eb1141xxxx`
(real data replaced by `xxxx` for security reason)

**Comment by beitler** [comment]
**tl;dr**, set `euiprefix` to 0 and you get what you want.

Thanks for posting this issue. What you are observing is one of the the ways Basic Station translates a 48-bit MAC address to at 64-bit EUI. First of all, any gateway ID where the higher (MSB) 16 bits are zero, is interpreted as a 48 bit MAC address. Now, there are multiple ways to convert this number to a 64-bit EUI which is controlled by the `euiprefix` field. If any of the lower 48 bits of `euiprefix` are non-zero, station applies an expansion by stuffing `ff:fe` in between the two halves of the MAC address to form a 64-bit number. If the lower 48 bits of `euiprefix` are zero, then the leading (MSB) 16 bits of the `euiprefix` are just pre-pended to the 48-bit MAC address, hence the name.
Some examples:
`"routerid":  "0000e8eb1141xxxx", "euiprefix": "::1" -> sation EUI: e8eb11fffe41xxxx`
`"routerid":  "0000e8eb1141xxxx", "euiprefix": "::0" -> sation EUI: 0000e8eb1141xxxx`
`"routerid":  "0000e8eb1141xxxx", "euiprefix": "4321::" -> sation EUI: 4321e8eb1141xxxx`

**Note**, that in Station config files you can write any 64 bit number in multiple ways, namely, as a decimal number, hex string or ID6 string. In these examples, the `euiprefix` is given in ID6 notation which is similar to the IPv6 address notation. You can read more about ID6 [here](https://doc.sm.tc/station/glossary.html#term-id6).

**Note**, that by leaving out the `routerid` field, Station will extract the `routerid` from the MAC address of the first network interface (eth0 in most cases). You can use the `euiprefix` as shown above to define how the extracted MAC address is converted to the Station EUI.

**Comment by beitler** [comment]
Passing the `-x` argument is equivalent to setting the field `euiprefix` in the station.conf file (the setting in station.conf has higher priority).

**Examples**:
Suppose the following station.conf:
```
    "station_conf": {
        "routerid":  "0000e8eb1141xxxx",
        /* Commented lines are ignored */
        /* "euiprefix":  "4321::0", */   
```
Then:
```
./station -x 1        -> Station EUI: e8eb11fffe41xxxx
./station -x 0        -> Station EUI: 0000e8eb1141xxxx
./station -x "4321::" -> Station EUI: 4321e8eb1141xxxx
```

**Comment by 4refr0nt** [comment]
Thank very much!

**Comment by rbaldwin13** [comment]
In an effort to improve our customer support experience and in recognition that our support backlog on GitHub has historically exceeded the capacity of our engineering team, we have taken the difficult decision to focus on the most contemporary issues reported and to close all others without confirmation of resolution.

Our belief is that issues which have remained unresolved and unaltered for extended periods of time are less likely to continue to pose a significant problem to the user than when they were originally filed. More contemporary issues however may still be relevant and hence are more appropriate to prioritize.

For those users who remain interested in resolution of a reported issue that was closed, we are encouraging usage of our developer portal forums [https://forum.lora-developers.semtech.com/] and commercial support portal [https://semtech.force.com/ldp/ldp_support] as the preferred avenues to receive support. We will continue to monitor the GitHub issue trackers as well, but want to encourage all users to take advantage of the increased community presence on the developer portal. For commercial customers, we highly recommend using the commercial support portal which is uniquely tailored to service such support requests.
